backend/capellacollab/projects/toolmodels/modelsources/git/handler/handler.py

```diff
@@ -59,9 +58,32 @@ def get_artifact_from_job_as_content(
     ) -> bytes:
         pass
 
-    @abstractmethod
+    @abc.abstractmethod
     async def get_file_from_repository(
-        self,
-        trusted_file_path: str,
-    ) -> requests.Response:
+        self, project_id: str, trusted_file_path: str
+    ) -> bytes:
         pass
+
+    async def get_file_from_repository_or_artifacts(
+        self, trusted_file_path: str, job_name: str | None
+    ) -> bytes:
+        project_id = await self.get_project_id_by_git_url()
+        try:
+            return await self.get_file_from_repository(
+                project_id, trusted_file_path
+            )
+        except (requests.HTTPError, Exception):
+            if job_name:
+                (
+                    _,
+                    (job_id, _),
+                ) = await self.get_last_job_run_id_for_git_model(
+                    job_name, project_id
+                )
+                return self.get_artifact_from_job_as_content(
+                    project_id, job_id, trusted_file_path
+                )
```

Comment:
I prefer here to simply put a `pass` into the except block. This decreases the nesting here and usually this increase the readability.
In addition, I would not put `Exception` in here as `requests.HTTPError` and `GitRepositoryFileNotFoundError` should be the only once that can be thrown or?

Reply:
yes I think that should be the only two possibilities